pkg/controller/pxc/users.go

```diff
@@ -182,8 +168,11 @@ func (r *ReconcilePerconaXtraDBCluster) manageSysUsers(cr *api.PerconaXtraDBClus
 }
 
 func (r *ReconcilePerconaXtraDBCluster) syncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) error {
+	if cr.Status.Status != api.AppStateReady || cr.Status.ProxySQL.Status != api.AppStateReady {
+		return nil
+	}
 	// sync users if ProxySql enabled
-	if cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled && cr.Spec.ProxySQL.Size > 0 {
+	if cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled {
```

Comment:
i think it's worth to rewrite 
```
if cr.Spec.ProxySQL == nil || !cr.Spec.ProxySQL.Enabled{
        return nil
}
```

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/commit/e381ed0a9d2b632fd61bb6df5c24749acec72d3f